lib/models/device.coffee

```diff
@@ -1121,10 +1121,10 @@ exports.getDeviceUrl = (uuid, callback) ->
 		if not hasDeviceUrl
 			throw new Error("Device is not web accessible: #{uuid}")
 
-		return configModel.getAll().get('deviceUrlsBase')
-	.then (deviceUrlsBase) ->
-		return "https://#{uuid}.#{deviceUrlsBase}"
-	.nodeify(callback)
+		configModel.getAll().get('deviceUrlsBase').then (deviceUrlsBase) ->
+			exports.get(uuid).get('uuid').then (uuid) ->
+				return "https://#{uuid}.#{deviceUrlsBase}"
+			.nodeify(callback)
```

Comment:
This `.nodeify()` call should be more indented to the left, to match line 1120, otherwise, this means we're only "nodeifying" the most inner promise.